### ISSUE TITLE
repeat epoch

### ISSUE THREAD

**Issue by Ontheroad123** [fastai/fastai#2512]
there are my code：
learn = cnn_learner(data,models.resnet101,metrics=error_rate, wd= wd)
plt.switch_backend('Agg')

def newtrain():
        lr = 1e-1
        learn.fit_one_cycle(10, slice(lr), pct_start=0.8, callbacks=[SaveModelCallback(learn, every='epoch', name = 'stage1')])
        learn.recorder.plot_lr()
        learn.save('stage-1')
        lr_find(learn)
        plt.savefig("liver_stage1.png")
newtrain()
but i find it run repeat:
epoch     train_loss  valid_loss  error_rate  time    
epoch     train_loss  valid_loss  error_rate  time    <00:00]
0         0.801805    0.629043    0.199278    03:57                     
0         0.828107    0.611591    0.199278    03:57              
1         0.645031    0.502624    0.179503    03:51                     
1         0.621404    0.509897    0.179503    03:51              
2         0.562800    0.447399    0.160953    03:52                     
2         0.562270    0.445950    0.160953    03:52              
3         0.496844    0.421863    0.151341    03:52                     
3         0.515444    0.422395    0.151341    03:53 
i have test the code, the problem in this code:
learn.fit_one_cycle(10, slice(lr), pct_start=0.8, callbacks=[SaveModelCallback(learn, every='epoch', name = 'stage1')]),but i don't how to solve it

**Comment by sgugger** [comment]
I don't understand what your problem is. Could you explain it a bit better? Also make sure to put all your code between ``` so that we can see the formatting.

**Comment by Ontheroad123** [comment]
all code
'''
from fastai import *
from fastai.vision import *
from fastai.callbacks.hooks import *
from fastai.distributed import *
import argparse
from fastai.callbacks import *
from sklearn.metrics import mean_absolute_error,r2_score,mean_absolute_error
from fastai.metrics import error_rate
np.random.seed(42)

parser = argparse.ArgumentParser()
parser.add_argument("--local_rank", type=int)
args = parser.parse_args()
torch.cuda.set_device(args.local_rank)
torch.distributed.init_process_group(backend='nccl', init_method='env://')

DATA_PATH = '***'
data = ImageDataBunch.from_folder(DATA_PATH, train=".", valid_pct=0.2, ds_tfms=get_transforms(), size=224, bs=64, num_workers=4).normalize(imagenet_stats)
#data.show_batch(rows=3, figsize=(15,15))

wd=1e-2
learn = cnn_learner(data,models.resnet101,metrics=error_rate, wd= wd)
plt.switch_backend('Agg')
#learn.lr_find()
# learn.recorder.plot()
# plt.savefig("ori.png")

def newtrain():
        lr = 1e-3
        learn.fit_one_cycle(20, slice(lr), pct_start=0.8, callbacks=[SaveModelCallback(learn, every='epoch', name = 'stage1', monitor='dice')])
        lr_find(learn)
        learn.recorder.plot_lr()
        learn.save('stage-1')
        plt.savefig("liver_stage1.png")
newtrain()
'''
and here are my training step:
'''
epoch     train_loss  valid_loss  error_rate  time    
epoch     train_loss  valid_loss  error_rate  time    ---------------------------------------------------| 0.00% [0/510 00:00<00:00]
0         0.900816    0.665730    0.204910    03:55                                                                                            
0         0.853243    0.636195    0.204910    03:56                                                                                     
1         0.713103    0.548370    0.182748    03:51                                                                                            
1         0.707130    0.548361    0.182748    03:52                                                                                     
2         0.611245    0.480037    0.173258    03:51                                                                                            
2         0.613976    0.499895    0.173258    03:50     -------------------------------------------------| 0.00% [0/510 00:00<00:00]    
3         0.547943    0.450105    0.164503    03:51                                                                                            
3         0.555117    0.454073    0.164503    03:50                                                                                     
4         0.525317    0.431651    0.157830    03:51                                                                                            
4         0.520752    0.429502    0.157830    03:52                                                                                     
5         0.503877    0.402140    0.147545    03:51                                                                                            
5         0.485039    0.409282    0.147545    03:50                                                                                     
6         0.471035    0.384446    0.138545    03:51                                                                                            
6         0.437211    0.386524    0.138545    03:52                                                                                     
7         0.441435    0.367307    0.129791    03:52                                                                                            
7         0.443754    0.368157    0.129791    03:51                                                                                     
8         0.441576    0.344207    0.122873    03:53                                                                                            
8         0.421347    0.345141    0.122873    03:52                                                                                     
9         0.413469    0.336272    0.119995    03:50                                                                                             
9         0.404962    0.337356    0.119995    03:50                                                                                      
10        0.392405    0.316121    0.114118    03:52                                                                                             
10        0.389751    0.327623    0.114118    03:52                                                                                      
11        0.388876    0.318401    0.114301    03:51                                                                                             
11        0.388485    0.328782    0.114301    03:52                                                                                      
12        0.393110    0.303263    0.105730    03:51                                                                                             
12        0.365484    0.308623    0.105730    03:52                                                                                      
13        0.382581    0.298208    0.104690    03:50                                                                                             
13        0.348877    0.303230    0.104690    03:51                                                                                      
14        0.355841    0.294234    0.098445    03:50                                                                                             
14        0.348680    0.289573    0.098445    03:51                                                                                      
15        0.346630    0.286972    0.099180    03:51                                                                                             
15        0.340996    0.284068    0.099180    03:50                                                                                      
16        0.333279    0.280358    0.095506    03:50                                                                                             
16        0.362145    0.278046    0.095506    03:50                                                                                      
17        0.339084    0.270761    0.095139    03:50                                                                                             
17        0.329470    0.277648    0.095139    03:50                                                                                      
18        0.329258    0.265324    0.091282    03:50                                                                                             
18        0.335396    0.267863    0.091282    03:51                                                                                      
19        0.310762    0.264930    0.090731    03:50                                                                                             
19        0.318840    0.264458    0.090731    03:50                                                                                      
epoch     train_loss  valid_loss  error_rate  time    
epoch     train_loss  valid_loss  error_rate  time    --------------------------------------------------| 0.00% [0/510 00:00<00:00]
0         1.176450    #na#        00:35                                                                                                     
0         0.868749    #na#        00:35        
'''
it looks like it trains repeat,every epoch has twice

**Comment by sgugger** [comment]
This is because you are training on multiple GPUs without using fastai distributed module, so you see the progress bar of each GPU. Your epochs are not repeated, no.
